noxfile.py

```diff
@@ -0,0 +1,49 @@
+import nox
+from nox.sessions import Session
+
+nox.options.default_venv_backend = "uv"
+nox.options.reuse_venv = True
+
+PYTHON_VERSIONS = ["3.8", "3.9", "3.10", "3.11", "3.12"]
+
+
+def run_common(session: Session, coverage_threshold: float) -> None:
+    session.install("-r", "requirements-dev.txt")
+
+    session.run(
+        "pytest",
+        "tests",
+        "--cov=narwhals",
+        "--cov=tests",
+        f"--cov-fail-under={coverage_threshold}",
+        "--runslow",
+    )
+    session.run("pytest", "narwhals", "--doctest-modules")
+
+
+@nox.session(python=PYTHON_VERSIONS)  # type: ignore[misc]
```

Comment:
type ignore is due to the following error raised in pre-commit:

> noxfile.py:24: error: Untyped decorator makes function "pytest_coverage" untyped  [misc]

Same for all the decorators below